### PR TITLE
router refactoring

### PR DESCRIPTION
**Pull request by nicpuddu**

### CODE REVIEW

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by gidesan** [comment]
retest this please

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is well

File: src/ui/app/App.js [hunk 1]
```diff
@@ -174,124 +178,188 @@ import CreateFolderPage from 'ui/file-browser/add/CreateFolderPage';
...
+  } else if (!username && currentRoute !== ROUTE_HOME) {
+    return <Redirect to={ROUTE_HOME} />;
   }
 
   return (
     <Fragment>
       <ToastsContainer />
       {!keycloak.enabled || keycloak.authenticated
-        ? getRouteComponent(route, keycloak)
+        ? getRouteComponent(keycloak)
```
Comment (gidesan) [inline src/ui/app/App.js]: the keycloak flag seems to not be used in the `getRouteComponent` method

File: Apps.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+several different items need to be exported by the App so that they can be consumed by the app builder.
+
+Each paragraph reports in angle brackets the name of the object that the App itself need exporting
+
+## ID <id>
+the unique identifier for the app:
+
+i.e. `cms`
+
+## Routing
```
Comment (gidesan) [inline Apps.md]: is the routing paragraph still needed after the removal of @entando/router?

**Comment by gidesan** [comment]
I don't want to spend too much time on this story because it needs to be quickly merged, but before merging I want to share some notes:

- having some “selector creators” instead of normal selector (to overcome the removal of route parameter in redux state) can be misleading
- as an alternative, we could have used react-router-redux and have a better retro-compatibility with our code, avoid selector creators and keep using removed selectors like getParams
- we could also have kept gotoRoute with a different implementation (a simple call to history.push)

**Comment by entando-service** [comment]
All is well

File: Apps.md (hunk 2 above)
Comment (nicpuddu) [reply Apps.md]: I'll be taking care of this later

Comment (gidesan) [reply Apps.md]: cool, thanks

**Pull request merged**